From b35b0b9335646ce49ca0f49309d43567a9f22b28 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20D=C3=B6singer?= <stefan@codeweavers.com>
Date: Wed, 10 Apr 2013 20:09:55 +0200
Subject: [PATCH 064/127] wined3d: Wait for the cs to finish before destroying
 the device
Reply-To: wine-devel <wine-devel@winehq.org>

---
 dlls/wined3d/device.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/dlls/wined3d/device.c b/dlls/wined3d/device.c
index bdb69be..6d27c57 100644
--- a/dlls/wined3d/device.c
+++ b/dlls/wined3d/device.c
@@ -978,6 +978,9 @@ HRESULT CDECL wined3d_device_uninit_3d(struct wined3d_device *device)
     if (!device->d3d_initialized)
         return WINED3DERR_INVALIDCALL;
 
+    if (wined3d_settings.cs_multithreaded)
+        device->cs->ops->finish(device->cs);
+
     /* Force making the context current again, to verify it is still valid
      * (workaround for broken drivers) */
     context_set_current(NULL);
-- 
1.8.1.5

